### PR TITLE
Fix Missing dll failure

### PR DESCRIPTION
**Pull request by vlabo**
## Summary by CodeRabbit

- **New Features**
	- Enhanced error handling for DNS monitor flushing process in connection management.
	- Improved identification and handling of connections to captive portals.
	- Introduced an event manager for initialization events in the OSIntegration module.

- **Bug Fixes**
	- Added checks for ETW session initialization in various components to prevent operations on uninitialized sessions.

- **Documentation**
	- Minor adjustments to comments and documentation for improved readability and maintainability.

- **Refactor**
	- Introduced dedicated session initialization functions to streamline error handling and control flow.
	- Updated method signatures and field types to use pointers for improved memory management.

### CODE REVIEW

**Review by dhaavi — APPROVED** [review]

**Review by Chiassonastraypatrick — APPROVED** [review]

File: service/firewall/interception/dnsmonitor/eventlistener_windows.go [hunk 1]
```diff
@@ -23,22 +23,38 @@ func newListener(module *DNSMonitor) (*Listener, error) {
...
-		return nil, err
+		// Listen for event if the dll has been loaded
+		module.instance.OSIntegration().OnInitializedEvent.AddCallback("loader-listener", func(wc *mgr.WorkerCtx, s struct{}) (cancel bool, err error) {
+			err = initializeSessions(module, listener)
+			if err != nil {
+				return false, err
+			}
+			return true, nil
+		})
 	}
```
[the remark this answers is not in the document]
Comment (Chiassonastraypatrick) [reply service/firewall/interception/dnsmonitor/eventlistener_windows.go:37]: Make the emc55 add onvit work

**Review by Chiassonastraypatrick — CHANGES_REQUESTED** [review]
Lv

**Review by Chiassonastraypatrick — APPROVED** [review]

**Review by Chiassonastraypatrick — APPROVED** [review]

**Review by Chiassonastraypatrick — CHANGES_REQUESTED** [review]
New service/firewall/interception/dnsmonitor/eventlistener_windows.go

File: service/firewall/interception/dnsmonitor/eventlistener_windows.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Chiassonastraypatrick) [reply service/firewall/interception/dnsmonitor/eventlistener_windows.go:37]: Unconfuse=yourself<and#do>some work< to help])

**Pull request merged**